tests/h/search/client_test.py

```diff
@@ -31,75 +44,139 @@ def test_conn_is_read_only(self):
             client.conn = "changed"
 
 
+class TestClientConnection(object):
+
+    @pytest.mark.parametrize('get_client,elasticsearch_dsl',
+        [(get_client, elasticsearch_dsl), (get_client_old_wrap, elasticsearch1_dsl)])
+    def test_client_connection_successful(self, client, index, User, get_client, elasticsearch_dsl):
+        # write
+        user = User(meta={'id': 1}, age=34)
+        user.save(index='user', using='es')
+        # read
+        user = User.get(id=1, index='user', using='es')
+        assert user.age == 34
+
+    @pytest.fixture
+    def User(self, request):
+        elasticsearch_dsl = request.getfixturevalue('elasticsearch_dsl')
+
+        class User(elasticsearch_dsl.DocType):
+            age = elasticsearch_dsl.Integer()
+
+        return User
+
+    @pytest.fixture
+    def client(self, request):
+        get_client = request.getfixturevalue('get_client')
+        elasticsearch_dsl = request.getfixturevalue('elasticsearch_dsl')
+        # create the connection
+        settings = {'es.url': 'http://localhost:9201',
+                    'es.host': 'http://localhost:9200',
+                    'es.index': 'myindex'}
+        client = get_client(settings)
+        elasticsearch_dsl.connections.connections.add_connection('es', client)
+        yield client
+        # delete connection
+        elasticsearch_dsl.connections.connections.remove_connection('es')
+
+    @pytest.fixture
+    def index(self, User, client, request):
+        elasticsearch_dsl = request.getfixturevalue('elasticsearch_dsl')
+        # create the index
+        index = elasticsearch_dsl.Index('user', using='es')
+        index.doc_type(User)
+        index.create()
+        yield index
+        # delete index
+        index.delete()
```

Comment:
We'll need test fixtures for the new Elasticsearch index to create the index before each test run and to delete the index after each test run, and to empty out the index in-between tests, but those should go in `tests/common/fixtures/elasticsearch.py` rather than in a particular test class. And I'm not sure that we need them until we're actually writing anything to the index (if we want a test to just test that the connection is available and working, can that just make some sort of Elasticsearch metadata query that doesn't require there to be an index?)

I don't think we should add an `es_client` fixture for the new Elasticsearch until we know that the code actually needs direct access to the client